### ISSUE TITLE
Dialog "Reset all changes" behaves incorrectly

### ISSUE THREAD

**Issue by kasal** [gitextensions/gitextensions#2203]
I have noticed a bug with the parameter hasExistingFiles of class FormResetChanges.

If there are no unstaged modified files, yet there are some staged ones, then the caller (StartResetChangesDialog) sets it to false, hence the dialog is locked to delete all new unstaged files as well.

If there are really no changes, then "Also delete new files" is locked cheked.
The idea is that without that, it would be a no-op.
But I think it would be safer and more consistent to leave it unchecked, so that the user has to check it.

To sum up, I propose to remove parameter hasExistingFiles altogether.

**Comment by EbenZhang** [comment]
I think it's smart/good to tick the `Also delete new files...`.

The bug is when there's staged modified files and only unstaged new files. In this case the reset all changes will tick the checkbox because it thought it's reseting only new files.

How about the following rules?

- If there're only new files, then disable and tick the checkbox automatically
- If there're new files and modified files(both staged and unstaged), then enable the checkbox and leave it unticked so the user can decide what to do
- If there's no new file,  then disable and untick the checkbox  automatically

**Comment by KindDragon** [comment]
We use dialog in three cases:
1. Reset all changes (including submodule)
2. Reset unStaged changes
3. Reset selected changes in commit dialog

We talking now about 1st and 2nd case?

**Comment by EbenZhang** [comment]
The 1st one. When you have only new files in the unstaged list and modified files in the staged list, in this scenario if you press the reset all changes button, GitExtension will tick the checkbox because it thought it's reseting only new files.
